File: 🟡 Medium/50 - Pow(x, n)/_50Pow(x, n).js
/**
 * @param {number} x
 * @param {number} n
 * @return {number}
 */
var myPow = function (x, n) {
    let ans = 1;

    if (n < 0) {
        x = 1 / x;
        n = Math.abs(n);
    }

    while (n > 0) {
        // LSB = 1 (Then multiply your answer with x)
        if ((n & 1) != 0) {
            ans *= x;
        }

        x *= x;
        n = Math.floor(n / 2);
    }

    return ans;
};

File: 🟡 Medium/50 - Pow(x, n)/_50Pow(x, n).c
long absolute(long power) {
    if (power < 0) {
        return power * (-1);
    }else{
        return power;
    }
}

double myPow(double x, int n) {
    double ans = 1.0;
    long power = n;

    if (n < 0) {
        x = 1 / x;
        power = absolute(power);
    }

    while (power > 0) {
        // LSB = 1 (Then multiply your answer with x)
        if ((power & 1) != 0) {
            ans *= x;
        }

        x *= x;
        power = power >> 1;
    }

    return ans;
}